lib/github-pages-health-check/domain.rb

```diff
@@ -0,0 +1,202 @@
+module GitHubPages
+  module HealthCheck
+    class Domain < GitHubPages::HealthCheck::Checkable
```

Comment:
You probably only need `Checkable` here, but more a style thing than anything about correctness.